### ISSUE TITLE
az containerapp create resets images for existing apps

### ISSUE THREAD

**Issue by aarondandy** [microsoft/azure-container-apps#917]
### This issue is a: (mark with an x)
- [x] bug report -> please search issues before submitting
- [x] documentation issue or request
- [ ] regression (a behavior that used to work and stopped in a new release)

### Issue description

Calls to `az containerapp create` without an explicit `--image` argument change the image of an existing container app to a k8s hello world image. We use this in our infrastructure scripts to set up container apps, configure their secrets and environment variables, and configure access with managed identities. We handle deploying revisions and images separately as part of a different deployment process. Ideally we would be able to just use `az containerapp create` to ensure our container app exists in a desired state before making further calls to `containerapp secret set --secrets`, `containerapp update --set-env-vars`, `containerapp registry set`, or whatever else we have going on for our apps.

### Workaround

We invoke `az containerapp show` to decide if a call to `az container app create` is required.

### Steps to reproduce
1. Invoke `az containerapp create --name .. --environment ..` as part of idempotent infrastructure automation, omitting `--image`
2. Invoke `az containerapp update --image` to deploy the application as part of our deployment automation
3. Invoke `az containerapp create --name .. --environment ..` as part of idempotent infrastructure automation, omitting `--image`

**Expected behavior** [What you expected to happen.]

Our application is still running.

**Actual behavior** [What actually happened.]

The k8s hello world sample has replaced our application.

### Additional context

Nerds can debate idempotent vs effectively idempotent, if `--image` is implicitly included in what would be considered idempotent on a `create` invocation or not, and if Azure CLI even should be idempotent. I would like to keep any discussion human centric and focused on how a call to `create` replacing an existing and running app with a Hello World image would be surprising behavior that would be problematic for users just trying to get through their day.

Maybe `az containerapp up` has the desired intent, but it requires an image to be specified, which we handle during application deployment.

**Comment by torosent** [comment]
Hi, A `create` is doing a PUT operation and `update` is doing a PATCH.  The put will overwrite the existing values.

**Comment by aarondandy** [comment]
Yeah that makes sense as the technical cause of the behavior but as a user of `az containerapp` the outcome of invoking this command is surprising and I don't have an alternative command to use for this case.

**Comment by aarondandy** [comment]
To add some context to this with respect to expectations set by existing parts of the Azure CLI, azure webapps will not overwrite an existing deployment with a "Hello World" equivalent when `az webapp create` is invoked for an existing webapp.

https://learn.microsoft.com/en-us/cli/azure/webapp?view=azure-cli-latest#az-webapp-create

**Comment by tjementum** [comment]
I kind of have the same problem, but I'm using Bicep.

1. I create the initial version of the container app using Bicep, for both staging and production. Like you, I would love to *not* specify a docker image, but the image is required, and so is the tag. So, I point it to the docker image, and use the `latest` tag.
2. Later I make an update to the container app and tag it with `v2`. I don't want to redeploy Bicep just to push a new version (it's slow and might push other infrastructure changes). So here I use `az containerapp update ... -image:xxx:v2`. So far so good.
3. But, the problem is that the next time I update the infrastructure (using Bicep) it will roll back this change to the `latest` tag.

It seems to me that there should be a way to update an existing app without changing the active version (both using Bicep or Azure CLI).

My workaround is to have a script that reads the tag of the current version and pass it as a parameter to Bicep. But this is complex, and there is a potential race condition.
